packages/js/components/src/experimental-select-control/menu.tsx

```diff
@@ -35,21 +34,28 @@ export const Menu = ( {
 	position = 'bottom right',
 	scrollIntoViewOnOpen = false,
 }: MenuProps ) => {
-	const [ boundingRect, setBoundingRect ] = useState< DOMRect >();
 	const selectControlMenuRef = useRef< HTMLDivElement >( null );
+	const popoverRef = useRef< HTMLDivElement >( null );
 
 	useLayoutEffect( () => {
-		if (
-			selectControlMenuRef.current?.parentElement &&
-			selectControlMenuRef.current?.parentElement.clientWidth > 0
-		) {
-			setBoundingRect(
-				selectControlMenuRef.current.parentElement.getBoundingClientRect()
+		const comboboxWrapper = selectControlMenuRef.current?.closest(
+			'.woocommerce-experimental-select-control__combo-box-wrapper'
+		);
+		const popoverContent =
+			popoverRef.current?.querySelector< HTMLDivElement >(
+				'.components-popover__content'
 			);
+		if ( comboboxWrapper && comboboxWrapper?.clientWidth > 0 ) {
+			if ( popoverContent ) {
+				popoverContent.style.width = `${
+					comboboxWrapper.getBoundingClientRect().width
+				}px`;
+			}
 		}
 	}, [
 		selectControlMenuRef.current,
 		selectControlMenuRef.current?.clientWidth,
+		popoverRef.current,
```

Comment:
There's an ESLint message regarding this line. Is this really needed?

`[react-hooks/exhaustive-deps] React Hook useLayoutEffect has unnecessary dependencies: 'popoverRef.current' and 'selectControlMenuRef.current'. Either exclude them or remove the dependency array. Mutable values like 'selectControlMenuRef.current' aren't valid dependencies because mutating them doesn't re-render the component.`

Reply:
Uhm! I think this is a false positive here since what is in fact a mutable var is `popoverRef` and `selectControlMenuRef` but that is not the case for the `current` prop within both mutable vars because current does not maintain the same reference as it's always a new value.